test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,348 @@
+}
+
+TEST_F(HystrixSinkTest, AddAndRemoveClusters) {
+  InSequence s;
+
+  // Arbitrary numbers for testing. Make sure error > timeout.
+  uint64_t success_step = 7;
+  uint64_t error_step = 17;
+  uint64_t timeout_step = 1;
+  uint64_t timeout_retry_step = 2;
+  uint64_t rejected_step = 8;
+
+  uint64_t success_step2 = 3;
+  uint64_t error_4xx_step2 = 1;
+  uint64_t error_4xx_retry_step2 = 2;
+  uint64_t error_5xx_step2 = 3;
+  uint64_t error_5xx_retry_step2 = 4;
+  uint64_t timeout_step2 = 3;
+  uint64_t timeout_retry_step2 = 1;
+  uint64_t rejected_step2 = 5;
+
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+
+  // Add new cluster.
+  const std::string cluster2_name{"test_cluster2"};
+  ClusterTestInfo cluster2(cluster2_name);
+  cluster_map_.emplace(cluster2_name, cluster2.cluster_);
+  // Redefining since cluster_map_ is returned by value.
+  ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+
+  // Generate data to both clusters.
+  sink_->flush(source_);
+  for (uint64_t i = 0; i < (window_size_ + 1); i++) {
+    buffer.drain(buffer.length());
+    // Cluster 1
+    ON_CALL(cluster1_.error_5xx_counter_, value()).WillByDefault(Return((i + 1) * 17));
+    ON_CALL(cluster1_.success_counter_, value()).WillByDefault(Return((i + 1) * 7));
+    cluster1_.cluster_info_->stats().upstream_rq_timeout_.add(1);
+    cluster1_.cluster_info_->stats().upstream_rq_per_try_timeout_.add(2);
+    cluster1_.cluster_info_->stats().upstream_rq_pending_overflow_.add(8);
+
+    // Cluster 2
+    ON_CALL(cluster2.error_5xx_counter_, value()).WillByDefault(Return((i + 1) * 1));
+    ON_CALL(cluster2.retry_5xx_counter_, value()).WillByDefault(Return((i + 1) * 2));
+    ON_CALL(cluster2.error_4xx_counter_, value()).WillByDefault(Return((i + 1) * 3));
+    ON_CALL(cluster2.retry_4xx_counter_, value()).WillByDefault(Return((i + 1) * 4));
+    ON_CALL(cluster2.success_counter_, value()).WillByDefault(Return((i + 1) * 3));
+    cluster2.cluster_info_->stats().upstream_rq_timeout_.add(3);
+    cluster2.cluster_info_->stats().upstream_rq_per_try_timeout_.add(1);
+    cluster2.cluster_info_->stats().upstream_rq_pending_overflow_.add(5);
+
+    sink_->flush(source_);
+  }
+
+  std::unordered_map<std::string, std::string> cluster_message_map =
+      buildClusterMap(TestUtility::bufferToString(buffer));
+  ASSERT_NE(cluster_message_map.find(cluster1_name_), cluster_message_map.end());
+  ASSERT_NE(cluster_message_map.find(cluster2_name), cluster_message_map.end());
+
+  // Check stream format and data.
+  validate_results(cluster_message_map[cluster1_name_], success_step, error_step, timeout_step,
+                   timeout_retry_step, rejected_step, window_size_);
+  validate_results(cluster_message_map[cluster2_name], success_step2,
+                   error_4xx_step2 + error_4xx_retry_step2 + error_5xx_step2 +
+                       error_5xx_retry_step2,
```

Comment:
what's the reason not to just make a single constant for the sum?

Reply:
Need each one to set values to counters, somehow forgot to use these after defining them...